cmd/vcert/result_writer.go

```diff
@@ -117,6 +122,91 @@ func (o *Output) AsPKCS12(c *Config) ([]byte, error) {
 	return bytes, nil
 }
 
+func (o *Output) AsJKS(c *Config) ([]byte, error) {
+
+	if len(o.Certificate) == 0 || len(o.PrivateKey) == 0 {
+		return nil, fmt.Errorf("at least certificate and private key are required")
+	}
+
+	//getting the certificate and adding its bytes to the chain of certificates in bytes
+	p, _ := pem.Decode([]byte(o.Certificate))
+	if p == nil || p.Type != "CERTIFICATE" {
+		return nil, fmt.Errorf("certificate parse error(1)")
+	}
+	var chainCertInBytes = p.Bytes
+
+	// getting each one of the certificates in the chain of certificates and adding their bytes to the chain of certificates in bytes
+	for _, chainCert := range o.Chain {
+		crt, _ := pem.Decode([]byte(chainCert))
+		chainCertInBytes = append(chainCertInBytes, crt.Bytes...)
+	}
+
+	// getting the bytes of the PK
+	p, _ = pem.Decode([]byte(o.PrivateKey))
+	if p == nil {
+		return nil, fmt.Errorf("missing private key PEM")
+	}
+	var privDER []byte
+	if x509.IsEncryptedPEMBlock(p) {
+		privDERDecrypted, err := x509.DecryptPEMBlock(p, []byte(c.KeyPassword))
+		if err != nil {
+			return nil, fmt.Errorf("private key PEM decryption error: %s", err)
+		}
+
+		privDER = privDERDecrypted
+	} else {
+		privDER = p.Bytes
+	}
+
+	//creating a JKS
+	keyStore := keystore.New()
+
+	//creating a PK entry
+	pkeIn := keystore.PrivateKeyEntry{
+		CreationTime: time.Now(),
+		PrivateKey:   privDER,
+		CertificateChain: []keystore.Certificate{
+			{
+				Type:    "X509",
+				Content: chainCertInBytes,
+			},
```

Comment:
I think this may be expecting one `keystore.Certificate` object per certificate in the chain (EE + all the CAs) rather than just one `keystore.Certificate` with the bytes of all of the certificates in the chain for its `Content`.

Reply:
you're correct, I've update that code to add each certificate individually instead of the chain of their bytes

Reply:
done!